src/ucs/datastruct/usage_tracker.h

```diff
@@ -0,0 +1,191 @@
+ *
+ * @return UCS_OK if successful, or an error code as defined by
+ * @ref ucs_status_t otherwise.
+ */
+ucs_status_t
+ucs_usage_tracker_create(unsigned ticks_per_flush, unsigned active_capacity,
+                         double active_thresh, unsigned eject_thresh,
+                         ucs_usage_tracker_flush_cb_t cb, void *opaque,
+                         ucs_usage_tracker_h *usage_tracker_p);
+
+
+/**
+ * @brief Destroys a Usage Tracker object.
+ *
+ * @param [in] usage_tracker  Handle to the Usage Tracker object.
+ */
+void ucs_usage_tracker_destroy(ucs_usage_tracker_h usage_tracker);
+
+
+/**
+ * @brief Ticks the usage tracker.
+ *
+ * Triggers increment of hit counters for all LRU entries.
+ * Each @ref ucs_usage_tracker_t::ticks_per_flush ticks, a flush operation is
+ * performed.
+ *
+ * @param [in] usage_tracker  Handle to the Usage Tracker object.
+ */
+void ucs_usage_tracker_tick(ucs_usage_tracker_h usage_tracker);
+
+
+/**
+ * @brief Get most active entries.
+ *
+ * @param [in]  usage_tracker  Handle to the Usage Tracker object.
+ * @param [out] state          Filled with a list of most active entries.
+ */
+//flush_cb
+
+
+/**
+ * @brief Update an entry with RX usage data.
+ *
+ * @param [in]  usage_tracker  Handle to the Usage Tracker object.
+ * @param [in]  key            Key to insert.
+ * @param [in]  score          RX score of the entry.
+ *
+ * @return Ejected entry in case there is no room in the active list, or NULL
+ * otherwise.
+ */
+void *ucs_usage_tracker_push_rx(ucs_usage_tracker_h usage_tracker, void *key,
+                                size_t score);
+
+
+/**
+ * @brief Get score of a specific entry.
+ *
+ * @param [in]  usage_tracker  Handle to the Usage Tracker object.
+ * @param [in]  key            Key of the entry.
+ * @param [out] score_p        Filled with the requested entry's score.
+ *
+ * @return UCS_OK if successful, or an error code as defined by
+ * @ref ucs_status_t otherwise.
+ */
+ucs_status_t ucs_usage_tracker_get_score(ucs_usage_tracker_h usage_tracker,
+                                         void *key, size_t *score_p);
+
+
+/**
+ * @brief Remove an entry from Usage Tracker.
+ *
+ * @param [in]  usage_tracker  Handle to the Usage Tracker object.
+ * @param [in]  key            Key of the entry to be removed.
+ *
+ * @return UCS_OK if successful, or an error code as defined by
+ * @ref ucs_status_t otherwise.
+ */
+ucs_status_t
+ucs_usage_tracker_remove(ucs_usage_tracker_h usage_tracker, void *key);
+
+
+/**
+ * @brief Add an entry to the Usage Tracker.
+ *
+ * @param [in]  usage_tracker  Handle to the Usage Tracker object.
+ * @param [in]  key            Key of the entry to be added.
+ */
+static UCS_F_ALWAYS_INLINE void
+ucs_usage_tracker_add(ucs_usage_tracker_h usage_tracker, void *key)
+{
+    ucs_lru_put(usage_tracker->lru, key);
+}
+
+
+static UCS_F_ALWAYS_INLINE size_t
+ucs_usage_tracker_score(const ucs_usage_tracker_element_t *item)

```

Comment:
what is the difference between ucs_usage_tracker_score and ucs_usage_tracker_get_score?

Reply:
ucs_usage_tracker_get_score is an API which receives a key.
ucs_usage_tracker_score is a helper function for internal use.

Reply:
they are both in the H file, so seems both are API